File: contains_duplicate/contains_duplicate.py
def containsDuplicate(nums):
    seen = set()
    for num in nums:
        if num in seen:
            return True
        seen.add(num)
    return False

def containsDuplicateOneLine(nums):
    return len(nums) != len(set(nums))

# Example usage
nums = [1, 2, 3, 4, 5, 1]
result = containsDuplicate(nums)
print("Contains duplicate:", result)  # Output: True

oneliner_result = containsDuplicateOneLine(nums)
print("Contains duplicate (one-liner):", oneliner_result)  # Output:

File: contains_duplicate/contains_duplicate.ts
export {};

function containsDuplicate(nums: number[]): boolean {
  const seen = new Set<number>();
  for (const num of nums) {
    if (seen.has(num)) {
      return true;
    }
    seen.add(num);
  };
  return false;
}

function containsDuplicateOneLine(nums: number[]): boolean {
  return nums.length !== new Set(nums).size;
}

const nums: number[] = [1, 2, 3, 4, 5, 1];
const result: boolean = containsDuplicate(nums);
console.log(`Does the array contain duplicates? ${result}`); // Output: true

const onelinerResult: boolean = containsDuplicateOneLine(nums);
console.log(`Does the array contain duplicates (one-liner)? ${onelinerResult}`); // Output: true
